pkg/dockerregistry/server/signaturedispatcher_test.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	testImage, err := registrytest.NewImageForManifest("user/app", registrytest.SampleImageManifestSchema1, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	testImage.DockerImageManifest = ""
+	testImage.Signatures = append(testImage.Signatures, testSignature)
+
+	client.AddReactor("get", "images", registrytest.GetFakeImageGetHandler(t, *testImage))
+
+	testImageStream := registrytest.TestNewImageStreamObject("user", "app", "latest", testImage.Name, testImage.DockerImageReference)
+	if testImageStream.Annotations == nil {
+		testImageStream.Annotations = make(map[string]string)
+	}
+	testImageStream.Annotations[imageapi.InsecureRepositoryAnnotation] = "true"
+	client.AddReactor("get", "imagestreams", imagetest.GetFakeImageStreamGetHandler(t, *testImageStream))
+
+	client.AddReactor("get", "imagestreamimages", registrytest.GetFakeImageStreamImageGetHandler(t, testImageStream, *testImage))
+
+	registryApp := handlers.NewApp(ctx, &configuration.Configuration{
+		Loglevel: "debug",
+		Auth: map[string]configuration.Parameters{
+			fakeAuthorizerName: {"realm": fakeAuthorizerName},
+		},
+		Storage: configuration.Storage{
+			"inmemory": configuration.Parameters{},
+			"cache": configuration.Parameters{
+				"blobdescriptor": "inmemory",
+			},
+			"delete": configuration.Parameters{
+				"enabled": true,
+			},
+		},
+		Middleware: map[string][]configuration.Middleware{
+			"registry":   {{Name: "openshift"}},
+			"repository": {{Name: "openshift"}},
+			"storage":    {{Name: "openshift"}},
+		},
+	})
+	RegisterSignatureHandler(registryApp)
+	registryServer := httptest.NewServer(registryApp)
+	defer registryServer.Close()
+
+	serverURL, err := url.Parse(registryServer.URL)
+	if err != nil {
+		t.Fatalf("error parsing server url: %v", err)
+	}
+	os.Setenv("DOCKER_REGISTRY_URL", serverURL.Host)
+
+	url := fmt.Sprintf("http://%s/extensions/v2/user/app/signatures/%s", serverURL.Host, testImage.Name)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		t.Errorf("failed to make request: %v", err)
+	}
+
+	httpclient := &http.Client{}
+	resp, err := httpclient.Do(req)
+	if err != nil {
+		t.Fatalf("failed to do the request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("unexpected response status: %v", resp.StatusCode)
+	}
+
+	content, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("failed to read body: %v", err)
+	}
+
+	if len(content) == 0 {
+		t.Fatalf("unexpected empty body")
+	}
+
+	var ans signatureList
+
+	if err := json.Unmarshal(content, &ans); err != nil {
+		t.Logf("received body: %v", string(content))
+		t.Fatalf("failed to parse body: %v", err)
+	}
+
+	if len(ans.Signatures) == 0 {
+		t.Fatalf("unexpected empty signature list")
+	}
+
+	if testSignature.Name != ans.Signatures[0].Name {
```

Comment:
Could `reflect.DeepEqual` be used here to compare not just Name, but Type and Content as well?